extensions/s3/src/main/java/io/deephaven/extensions/s3/S3CompletableOutputStream.java

```diff
@@ -344,8 +344,8 @@ private void completeMultipartUpload() throws IOException {
      */
     private void abortMultipartUpload() throws IOException {
         if (uploadId == null) {
-            throw new IllegalStateException("Cannot abort multipart upload for uri " + uri + " because upload ID " +
```

Comment:
This is a separate bug fix where we can hit this case if we fail on starting to write to S3. In that case, we will call abort without an uploadId, since uploadId is set on the first write.